components/ArticleCard.js

```diff
@@ -1,19 +1,88 @@
 import styled from "styled-components";
-import Link from "next/link";
+import { useRouter } from "next/router";
+import { useStore } from "../hooks/useStore";
 
-export default function ArticleCard({ id, text }) {
+export default function ArticleCard({
+  id,
+  url,
+  urlToImage,
+  title,
+  author,
+  description,
+  message,
+  alert,
+}) {
+  const router = useRouter();
+  const setArticle = useStore((state) => state.setArticle);
+
+  async function getFullContent() {
+    const response = await fetch(`/api/search/${id}`, {
+      method: "POST",
+      body: JSON.stringify({ url: url }),
+    });
+
+    const data = await response.json();
+    setArticle({ fullcontent: data });
+    router.push(`/${id}`);
+  }
   return (
-    <Link href={`/${id}`}>
-      <StyledCard>{text}</StyledCard>
-    </Link>
+    <li>
+      {alert ? (
+        alert === "error" ? (
+          <StyledError>{message}</StyledError>
+        ) : (
+          <StyledWarning>{message}</StyledWarning>
+        )
+      ) : (
+        <StyledCard>
+          <article>
+            <img
+              src={urlToImage}
+              alt="article image"
+              width="250"
+              height="150"
+            ></img>
+          </article>
+
+          <article>
+            <Title onClick={getFullContent}>{title}</Title>
+            <Author>BY {author}</Author>
+            <Description>{description}</Description>
+          </article>
```

Comment:
these are not two articles after another, maybe you can change the layout such that you don't need the article elements

Reply:
or use different types of elements, such as 'div'

Reply:
ok thanks!